Packages/ChatSecure/ChatSecure/Classes/CallKitManager/CallManager.swift

```diff
@@ -0,0 +1,433 @@
+		//                // make sure this JSON is in the format we expect
+		//                if let turnJson = try JSONSerialization.jsonObject(with: turnData, options: []) as? [String: Any] {
+		//                    let turnUser = turnJson["user"] as? String
+		//                    let turnPwd = turnJson["pwd"] as? String
+		//
+		//                    UserDefaults.standard.setValue(turnUser, forKey: Constants.keySaveTurnServerUser)
+		//                    UserDefaults.standard.setValue(turnPwd, forKey: Constants.keySaveTurnServerPWD)
+		//                    UserDefaults.standard.synchronize()
+		//                }
+		//            } catch let error as NSError {
+		//                print("Failed to load: \(error.localizedDescription)")
+		//            }
+		//
+		//            let hasVideo = callType == "video"
+		//            let isGroupCall = groupType == "group"
+		//            let callerName = isGroupCall ? groupName : username
+		//
+		//            reportIncomingCall(isCallGroup: isGroupCall,
+		//                               roomId: roomId,
+		//                               clientId: clientId,
+		//                               avatar: avatar,
+		//                               token: token,
+		//                               callerName: callerName ?? "",
+		//                               hasVideo: hasVideo,
+		//                               completion: completion)
+		//        }
+	}
+}
+
+extension CallManager {
+	// MARK: Incoming Calls
+	/// Use CXProvider to report the incoming call to the system
+	func reportIncomingCall(isCallGroup: Bool, roomId: String, clientId: String, avatar: String?, token: String?, callerName: String, hasVideo: Bool = true, completion: ((NSError?) -> Void)? = nil) {
+		// Construct a CXCallUpdate describing the incoming call, including the caller.
+		let update = CXCallUpdate()
+		update.remoteHandle = CXHandle(type: .phoneNumber, value: callerName)
+		update.hasVideo = hasVideo
+		let uuid = UUID()
+		// pre-heat the AVAudioSession
+		// OTAudioDeviceManager.setAudioDevice(OTDefaultAudioDevice.sharedInstance())
+		
+		// Report the incoming call to the system
+		provider.reportNewIncomingCall(with: uuid, update: update) { [weak self] error in
+			guard let self = self else { return }
+			/*
+			 Only add incoming call to the app's list of calls if the call was allowed (i.e. there was no error)
+			 since calls may be "denied" for various legitimate reasons. See CXErrorCodeIncomingCallError.
+			 */
+			if error == nil {
+				let call = CallBox(uuid: uuid, clientId: clientId)
+				call.clientName = callerName
+				call.roomId = Int64(roomId) ?? 0
+				call.groupToken = token
+				call.avatar = avatar
+				call.isCallGroup = isCallGroup
+				call.type = hasVideo ? .video : .audio
+				self.addCall(call)
+			}
+			
+			completion?(error as NSError?)
+		}
+	}
+	
+	func sendFakeAudioInterruptionNotificationToStartAudioResources() {
+		var userInfo = Dictionary<AnyHashable, Any>()
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/ad9f0bea03722cdeaf2e622393c8e11e8aa08592/Packages/ChatSecure/ChatSecure/Classes/CallKitManager/CallManager.swift#L244"></span>Shorthand syntactic sugar should be used, i.e. [String: Int] instead of Dictionary&lt;String, Int&gt;.<br />
<code>syntactic_sugar</code> <code>CallManager.swift:244</code></td>
    </tr>
  </tbody>
</table>